src/models/timeaxisheaderview.cpp

```diff
@@ -0,0 +1,257 @@
+#include "timeaxisheaderview.h"
+
+#include <QPainter>
+#include <QtMath>
+#include <QDebug>
+
+#include "filterandzoomstack.h"
+
+TimeAxisHeaderView::TimeAxisHeaderView(FilterAndZoomStack *filterAndZoomStack, Qt::Orientation orientation, QWidget *parent)
+    : QHeaderView(orientation, parent)
+    , m_filterAndZoomStack(filterAndZoomStack)
+{
+    setMinimumHeight(40);
+
+    connect(filterAndZoomStack, &FilterAndZoomStack::filterChanged, this, &TimeAxisHeaderView::onFilterChanged);
+    connect(filterAndZoomStack, &FilterAndZoomStack::zoomChanged, this, &TimeAxisHeaderView::onZoomChanged);
+}
+
+void TimeAxisHeaderView::setTimeRange(const Data::TimeRange &timeRange)
+{
+    m_timeRange = timeRange;
+    emit headerDataChanged(orientation(), 1, 1);
+}
+
+void TimeAxisHeaderView::onFilterChanged(const Data::FilterAction &filterAction)
+{
+    Q_UNUSED(filterAction)
+    emit headerDataChanged(orientation(), 1, 1);
+}
+
+void TimeAxisHeaderView::onZoomChanged(const Data::ZoomAction &zoomAction)
+{
+    Q_UNUSED(zoomAction)
+    emit headerDataChanged(orientation(), 1, 1);
+}
+
+void TimeAxisHeaderView::paintSection(QPainter *painter, const QRect &rect, int logicalIndex) const
+{
+    if (painter == nullptr) return;
+    if (logicalIndex != 1) return; // TODO: role based?
+
+    auto zoomTime = m_filterAndZoomStack->zoom().time;
+    if (!zoomTime.isValid())
+    {
+        zoomTime = m_timeRange; // full
+    }
+    if (!zoomTime.isValid())
+    {
+        return; // bailing out, no valid range to show
+    }
+    const double oneNanoSecond = 1.0e-9;
+    const double start = zoomTime.start * oneNanoSecond;
+    const double end = zoomTime.end * oneNanoSecond;
+
+    QBrush text(QColor(0,0,0));
+    painter->setBrush(text);
+
+    const double resolution = (end - start) / rect.width();
+    const auto xForTime = [rect, start, resolution](const double time) {
+        return rect.x() + qRound((time - start)/resolution);
+    };
+
+    const int fontSize = painter->fontMetrics().height();
+    const int endLabelWidth = painter->fontMetrics().horizontalAdvance(QLatin1String("-xXXXm"));
+    const int targetNbTicks = rect.width() / endLabelWidth;
+
+    qDebug() << __FUNCTION__ << start << end << targetNbTicks;
+    const PrefixTickLabels pfl(start, end, targetNbTicks);
+
+    painter->setPen(QColor(0,0,0));
+    if (pfl.hasPrefix())
+    {
+        const QString placeholder = QLatin1String("xxx");
```

Comment:
`const auto placeholder = QStringLiteral("xxx");`

your version allocates a temporary `QString` for the conversion from `QLatin1String`